drivers/cps-hid.c

```diff
@@ -666,6 +667,29 @@ static int cps_fix_report_desc(HIDDevice_t *pDev, HIDDesc_t *pDesc_arg) {
 		}
 	}
 
+	/* Fix for nominal power reporting getting clipped by LogMax. */
+	if ((pData=FindObject_with_ID_Node(pDesc_arg, 24 /* 0x18 */, USAGE_POW_CONFIG_ACTIVE_POWER))) {
+		upsdebugx(4, "Original Report Descriptor: ConfigActivePower "
+			"LogMin: %ld LogMax: %ld",
+			pData->LogMin, pData->LogMax);
+
+		/* Set a generous maximum that won't restrict UPS reporting.
+		 *
+		 * Current findings suggest that the values sent by the UPS are
+		 * accurate, but then get clipped by too strict LogMax thresholds:
+		 * https://github.com/networkupstools/nut/issues/2917#issuecomment-2832243477
+		 *
+		 * This could later be expanded to more advanced fiddling, where necessary.
+		 */
+		pData->LogMax = CPS_WATTAGE_LOGMAX;
```

Comment:
I think it should not be blanket-applied to every case, but if e.g. the original LogMax is smaller (so we do not truncate a 10KW UPS to 2KW either)? And perhaps if the value (like 510 in the practical case) is even smaller, if known at this point?

Reply:
Heh, was already at it while you commented, with the limited information we have perhaps just checking if smaller than reasonable is enough for now (until we learn more)? Let me know, will then add to NEWS and bump subdriver version.

Reply:
I thought about going the value route too, but I wasn't sure if we can access the value at this point as other descriptor fixes seem not to interact with the value within the function. A value-based fix would definitely also make sense, if we can realise it, e.g. check if value > logMax && logMax < reasonable ceiling => set logMax to reasonable ceiling.

Reply:
Yep, the change looks good. Checking for current value being under the reported LogMax is a cherry on top, and a bit of no-op (upon second thought) - we would anyway trim (or not) using the larger threshold value than originally reported.

Reply:
Thanks for double-checking! I'll also rename the constant to something less ambigous, I think... will get to the remaining changes hopefully later tonight.